pipeline-modules/deployment-service/aws/pipeline-config.yaml

```diff
@@ -174,29 +177,26 @@ template: |
       phases:
         pre_build:
           commands:
-            # installing jq v1.6
-            - curl -sLJO https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64
-            - mkdir {{workspace}}/bin && mv jq-linux64 {{workspace}}/bin/jq
-            - chmod +x {{workspace}}/bin/jq
-            - export PATH="{{workspace}}/bin:$PATH"
-            - |
-              echo "$git_repo: {{tag}}"
-              if [ ! -z "$git_repo" -a "$git_repo" != " " ]; then
-                  mkdir -p {{source_code_path}}
-                  git clone $git_repo --branch {{tag}} --single-branch {{source_code_path}}
-                  ls {{source_code_path}}/{{repo_dir}}/
-              fi
+            - echo "Cloning the pipeline modules..."
+            - mkdir -p {{pipeline_modules_path}}
+            - git clone {{pipeline_module_git}} --branch {{pipeline_module_git_rev}} --single-branch {{pipeline_modules_path}}
+
+            - {{ common_scripts_path }}/install-jq.sh
+
+            {% if artifact_type == 'GitCode' %}
```

Comment:
Just wondering. Instead of passing additional artifact_type input could we just use the existing variables like `git_username` or `$GIT_PASSWORD_ENCODED` to determine when to set the Git access? Or just check the protocol on the repository URL?

Reply:
I was kind of expecting that now that we are actually supporting more type of artifacts at deployment, that it would be a useful value to pass in for template writers.